lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -176,14 +166,35 @@ def set_request_tags!(trace, request_span, env, status, headers, response, origi
               request_span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_METHOD, env['REQUEST_METHOD'])
             end
 
+            url = parse_url(env, original_env)
+
             if request_span.get_tag(Tracing::Metadata::Ext::HTTP::TAG_URL).nil?
-              options = configuration[:quantize]
+              options = configuration[:quantize] || {}
+
+              # Quantization::HTTP.url base defaults to :show, but we are transitioning
+              options[:base] ||= :exclude
```

Comment:
This has to be done here, as doing it as a config default would be silently overridden when `:quantize` is manually set, which would be most surprising to the user.

Probably a general problem with such nested configuration keys.

Reply:
If I understand it correctly, this will be removed in the next major release, is that right?

If so, I suggest writing them down in the comment, so we can find it later.

Reply:
Actually, can you create a new file named `docs/NextReleaseUpgradeGuide.md` (name suggestion only), just add one section:
```
# Breaking changes

To aid with the next release process, please tag future breaking changes the inline code comment `# DEV-2.0:` prefix. 

## List of breaking changes

* `http.url` will change its default...
```

We've been meaning to do it, and it seems like your PR wants to have a breaking change in the next release.

Reply:
With this, I suggest adding the comment in this line with the prefix `# DEV-2.0:`.

Reply:
Good points! Will do.

Reply:
I think I'll do that in another PR, along with this suggestion: https://github.com/DataDog/dd-trace-rb/pull/2283#discussion_r975902685